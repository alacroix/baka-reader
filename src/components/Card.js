// @flow

import React from 'react';
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';

import appStyle from 'bakareader/src/appStyle';

const styles = StyleSheet.create({
  wrapper: {
    width: 160,
    height: 300,
    marginLeft: appStyle.grid.x2,
    marginRight: appStyle.grid.x2,
  },
  cover: {
    backgroundColor: '#ccc',
    shadowColor: 'rgba(0, 0, 0, 0.5)',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowRadius: 6,
    shadowOpacity: 1,
  },
  image: {
    width: 160,
    height: 240,
  },
  title: {
    marginTop: 10,
    fontSize: appStyle.font.size.large,
    textAlign: 'center',
    color: '#4a4a4a',
  },
});

type PropsType = {
  coverUri: any,
  onPress?: () => void,
  title: string
}

function Card(props: PropsType) {
  return (
    <TouchableOpacity activeOpacity={0.5} onPress={props.onPress} style={styles.wrapper}>
      <View style={styles.cover}>
        <Image source={props.coverUri} style={styles.image} />
      </View>
      <Text style={styles.title}>
        {props.title}
      </Text>
    </TouchableOpacity>
  );
}

Card.defaultProps = {
  onPress: () => {},
};

export default Card;
