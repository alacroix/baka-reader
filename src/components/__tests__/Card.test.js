import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Card from '../Card';

describe('Card', () => {
  it('should render the card', () => {
    const tree = renderer.create(
      <Card
        coverUri={require('bakareader/assets/images/dummy/cover1.jpg')}
        title="Card title"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
