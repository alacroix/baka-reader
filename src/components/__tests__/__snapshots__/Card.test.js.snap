// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render the card 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 300,
      "marginLeft": 16,
      "marginRight": 16,
      "opacity": 1,
      "width": 160,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "#ccc",
        "shadowColor": "rgba(0, 0, 0, 0.5)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 6,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 240,
          "width": 160,
        }
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#4a4a4a",
        "fontSize": 17,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    Card title
  </Text>
</View>
`;
